docs/build_process.rst

```diff
@@ -0,0 +1,362 @@
+a Koji task. Instead the orchestrator build tells it which Koji task
+ID to stream the filesystem tar archive from. Each worker build only
+streams the filesystem tar archive for the architecture it is running
+on.
+
+Arrangement version 3 (Koji build created by orchestrator)
+----------------------------------------------------------
+
+This arrangement builds on version 2. The ``koji_promote`` plugin,
+which was previously responsible for creating the Koji build, is
+replaced by these plugins:
+
+- worker build
+
+  * koji_upload
+
+- orchestrator build
+
+  * fetch_worker_metadata (see :ref:`Metadata Fragment Storage`)
+
+  * koji_import
+
+  * koji_tag_build
+
+Additionally the ``sendmail`` plugin now runs in the orchestrator
+build and not the worker build.
+
+.. _`Metadata Fragment Storage`:
+
+Metadata Fragment Storage
+-------------------------
+
+When creating a Koji Build using arrangement 3 and newer, the
+koji_import plugin needs to assemble Koji Build Metadata, including:
+
+- components installed in each builder image (worker builds and
+  orchestrator build)
+
+- components installed in each built image
+
+- information about each build host
+
+To assist the orchestrator build in assembling this (JSON) data, the
+worker builds gather information about their build hosts, builder
+images, and built images. They then need to pass this data to the
+orchestrator build. After creating the Koji Build, the orchestrator
+build must then free any resources used in passing the data.
+
+The method used for passing the data from the worker builds to the
+orchestrator build is to store it temporarily in a ConfigMap object in
+the worker cluster. Its name is stored in the OpenShift Build
+annotations for the worker build. To do this the worker cluster's
+"builder" service account needs permission to create ConfigMap
+objects.
+
+The orchestrator build collects the metadata fragments and assembles
+them together with the platform-neutral metadata in the koji_import
+plugin.
+
+The orchestrator build is then responsible for removing the OpenShift
+ConfigMap from the worker cluster. To do this, the worker cluster's
+"orchestrator" service account needs permission to get and delete
+ConfigMap objects.
+
+Arrangement version 4 (Multiple worker builds)
+----------------------------------------------
+
+This arrangement moves most of the Pulp integration work to the
+orchestrator build, allowing for multiple worker builds. Only the
+pulp_push plugin remains in the worker build.
+
+.. _group-manifests:
+
+A new plugin, group_manifests, creates a `manifest list`_ object in
+the registry, grouping together the image manifests from the worker
+builds.
+
+.. _`manifest list`: https://docs.docker.com/registry/spec/manifest-v2-2/#manifest-list
+
+.. graphviz:: images/arrangement-v4.dot
+   :caption: Orchestrator and worker builds (arrangement v4)
+
+Arrangement version 5 (ODCS Integration)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
I think this should use `---` for title, otherwise becomes a child of arrangement version 4.